timothyr/github-pom-version-scanner
# Scan every repository's pom.xml in Org

### What's new

Site is live at https://timothyr.github.io/github-pom-version-scanner/  

Scanning an org will display the `pom.xml` dependencies from every repository in the org

### Limitations

1. Only scans `pom.xml` in root folder. No nested pom files at this time
2. Rate limited by GitHub public API limits (See todo 1)
3. Will not cross reference versions from parent poms or other files

### Todo
In list of priority:  
1. Add a login to increase rate limit
2. Add ability to change baseurl for GitHub enterprise
3. Scan `pom.xml` in all folders (should be a checkbox, this is not always necessary and will require a lot more API calls)

### How to test
1. Go to https://timothyr.github.io/github-pom-version-scanner/  
2. Type in a GitHub org (Specifically one with Java repositories - otherwise there will not be any `pom.xml` file
3. Click scan and observe that all dependencies and their versions are listed
